engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/EngineJobExecutionActor.scala

```diff
@@ -99,6 +99,10 @@ class EngineJobExecutionActor(replyTo: ActorRef,
     } else callCachingParameters.mode
   }
 
+  //noinspection ActorMutableStateInspection
+  // The token dispenser from which this actor has most recently received a token, if any.
+  private var mostRecentTokenDispenser: Option[ActorRef] = None
```

Comment:
I don't fully understand the state machine, but I'd guess that this might flip once from `Option(jobRestartCheckTokenDispenserActor)` to `Option(jobExecutionTokenDispenserActor)` but not constantly flip-flop between the two. Is that correct?

Reply:
Yes exactly. 

If not restarting:

```
None => Option(jobExecutionTokenDispenserActor)
```

If restarting:

```
None => Option(jobRestartCheckTokenDispenserActor)
```
then if the job was not already complete:
```
Option(jobRestartCheckTokenDispenserActor) => Option(jobExecutionTokenDispenserActor)
```